testplan/runnable/interactive/http.py

```diff
@@ -46,6 +54,89 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _extract_cases_from_parametrization(param_entry: Dict) -> List[str]:
+    """
+    Given a parametrization entry, extracts the testcases.
+
+    :param param_entry: parametrization entry
+    :return: list of testcase names
+    """
+    cases = []
+    for entry in param_entry["entries"]:
+        if entry["category"] == ReportCategories.TESTCASE:
+            cases.append(entry["name"])
+    return cases
+
+
+def _extract_cases_from_suite(suite_entry: Dict) -> List[str]:
+    """
+    Given a testsuite entry, extracts the testcases.
+
+    :param suite_entry: testsuite entry
+    :return: list of testcase names
+    """
+    cases = []
+    for entry in suite_entry["entries"]:
+        if entry["category"] == ReportCategories.TESTCASE:
+            cases.append(entry["name"])
+        elif entry["category"] == ReportCategories.PARAMETRIZATION:
+            cases.extend(_extract_cases_from_parametrization(entry))
+    return cases
+
+
+def _extract_suites_cases(shallow_entry: Dict) -> Dict[str, List[str]]:
+    """
+    Given a shallow report entry, extracts the testsuites and testcases
+        to be run.
+
+    :param shallow_entry: holds entry name and category for all children
+    :return: mapping of testsuites to testcases to be run
+    """
+    category = shallow_entry["category"]
+    suites_cases = {}
+    if category == ReportCategories.MULTITEST:
+        for suite in shallow_entry["entries"]:
+            suites_cases[suite["name"]] = _extract_cases_from_suite(suite)
+    elif category == ReportCategories.TESTSUITE:
+        suites_cases[shallow_entry["name"]] = _extract_cases_from_suite(
+            shallow_entry
+        )
+    elif category == ReportCategories.PARAMETRIZATION:
+        suites_cases[
+            shallow_entry["parent_uids"][2]
+        ] = _extract_cases_from_parametrization(shallow_entry)
+    elif category == ReportCategories.TESTCASE:
+        suites_cases[shallow_entry["parent_uids"][2]] = [shallow_entry["name"]]
+    return suites_cases
+
+
+def _extract_entries(entry) -> Dict[str, Dict]:
```

Comment:
This function is to generate a "tree" of entries essentially to be used in setting runtime status for a selection of entries only.